tests/test_update.py

```diff
@@ -261,6 +269,47 @@ def test_effective_user(self, update):
         else:
             assert user is None
 
+    def test_effective_sender(self, update):
+        # Test that it's sometimes None per docstring
+        sender = update.effective_sender
+        if not (
+            update.channel_post is not None
+            or update.edited_channel_post is not None
+            or update.poll is not None
+            or update.chat_boost is not None
+            or update.removed_chat_boost is not None
+            or update.message_reaction_count is not None
+        ):
+            assert sender.id == 1
```

Comment:
I would have expected something like `assert isinstance(sender, Chat/User)` here making sure that we're getting the right attribute for the right use cases 🤔

Reply:
Right. this is why i intended to have two tests, although currently not working as you suggest..

`test_effective_sender_non_anonymous`: makes sure sender is either `User` or `None`
`test_effective_sender_anonymous`: makes sure sender is either `Chat`, `User` or `None`

i'll make them them more clear 😅